### PR TITLE
Storm 2 ElasticSearch

### PR DESCRIPTION
**Pull request by RotaruDan**

### CODE REVIEW

File: src/main/java/es/eucm/gleaner/realtime/RealTime.java [hunk 1]
```diff
@@ -34,7 +36,11 @@ private static StormTopology buildTopology(Map conf, String sessionId,
 		DBUtils.startRealtime(DBUtils.getMongoDB(conf), sessionId);
 
 		KafkaTopology kafkaTopology = new KafkaTopology(sessionId);
-		kafkaTopology.prepare(new MongoStateFactory(), zookeeperUrl);
+
+		String elasticsearchUrl = conf.get("elasticsearchUrl").toString();
+		EsConfig esConfig = new EsConfig(elasticsearchUrl, sessionId);
```
Comment (gorco) [inline src/main/java/es/eucm/gleaner/realtime/RealTime.java]: The name is esConfig (Elastic Search Config) but Mongo also use it?? XD

File: src/main/java/es/eucm/gleaner/realtime/RealTime.java [hunk 2]
```diff
@@ -89,11 +95,16 @@ public static void main(String[] args) {
...
+	// Storm flux function
 	public StormTopology getTopology(Map<String, Object> conf) {
-		DBUtils.startRealtime(DBUtils.getMongoDB(conf), conf.get("sessionId").toString());
-		KafkaTopology kafkaTopology = new KafkaTopology(conf.get("sessionId").toString());
-		kafkaTopology.prepare(new MongoStateFactory(), conf.get("zookeeperUrl").toString());
+		String sessionId = conf.get("sessionId").toString();
+		DBUtils.startRealtime(DBUtils.getMongoDB(conf), sessionId);
+		KafkaTopology kafkaTopology = new KafkaTopology(sessionId);
+		String elasticsearchUrl = conf.get("elasticsearchUrl").toString();
+		EsConfig esConfig = new EsConfig(elasticsearchUrl, sessionId);
```
Comment (gorco) [inline src/main/java/es/eucm/gleaner/realtime/RealTime.java]: The same

File: src/main/java/es/eucm/gleaner/realtime/states/MongoGameplayState.java [hunk 3]
```diff
@@ -18,35 +18,82 @@
...
+	}
+
+	public void prepare() {
+		if (esConfig == null) {
+			return;
+		}
+		try {
+			synchronized (EsState.class) {
+				if (EsState.client == null) {
+					EsState.client = new StormElasticSearchClient(esConfig)
```
Comment (gorco) [inline src/main/java/es/eucm/gleaner/realtime/states/MongoGameplayState.java]: The MongoGamePlayState with an instance of EsState??? Should exist a ElasticSearchGameplayState
and split the TridentStream as we spoke

**Comment by RotaruDan** [comment]
All comments addressed

File: src/main/java/es/eucm/gleaner/realtime/states/ESGameplayState.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	@Override
+	public void setProperty(String versionId, String gameplayId, String key,
+			Object value) {
+
+		try {
+			key = key.replace(".", "-");
+			UpdateRequest updateRequest = new UpdateRequest(resultsIndex,
+					"rage", gameplayId).doc(
```
Comment (gorco) [inline src/main/java/es/eucm/gleaner/realtime/states/ESGameplayState.java]: Create a constant "rage"

**Comment by gorco** [comment]
Great! 
Solve the last comment and mege the pr.

**Pull request merged**